### PR TITLE
Helpful dialog when creating Linux instances that require puppet roles

### PR DESCRIPTION
**Pull request by trainb0t**
Issue whereby if a deployment map is created and a puppet role is not specified, then IPTables
firewall rules are not updated between the NGINX load balancer and the AWS EC2 instance. The result
is all health checks passing but not traffic get's routed to the upstream. This can be a little
confusing for people creating deployment maps for the first time using Linux instances.

### CODE REVIEW

File: client/app/configuration/deployment-maps/DeploymentMapTargetController.js [hunk 1]
```diff
@@ -267,20 +247,39 @@ angular.module('EnvironmentManager.configuration').controller('DeploymentMapTarg
...
     }
 
-    function validateDataForSave() {
-      var valid = { result: true };
+    function validateDataForPuppetRole() {
+      if (vm.target.ASG.LaunchConfig.AMIPlatform == 'Linux' && typeof vm.target.ASG.LaunchConfig.PuppetRole == 'undefined') {
+        return modal.confirmation({
+          title: 'Linux AMI/Puppet Role Warning',
+          message: 'It looks like you have chosen a linux AMI and not specified a puppet role. This can lead to problems later on when you deploy to an environment. Please ask on the #linux-infra channel in slack if you need help defining one. Do you want to continue?'
+        });
```
Comment (jake-cross) [inline client/app/configuration/deployment-maps/DeploymentMapTargetController.js]: It would be nice to have this link directly to the Slack channel (like in `DeploymentDetailsModalController.js` vm.deploymentSupportSlack = "slack://channel?id=C1HCFJU67&team=T060F53FW"; )

File: client/app/configuration/deployment-maps/DeploymentMapTargetController.js [hunk 2]
```diff
@@ -267,20 +247,39 @@ angular.module('EnvironmentManager.configuration').controller('DeploymentMapTarg
       return _.map(deploymentMap.Value.DeploymentTarget, 'ServerRoleName');
     }
 
-    function validateDataForSave() {
-      var valid = { result: true };
+    function validateDataForPuppetRole() {
+      if (vm.target.ASG.LaunchConfig.AMIPlatform == 'Linux' && typeof vm.target.ASG.LaunchConfig.PuppetRole == 'undefined') {
```
Comment (jake-cross) [inline client/app/configuration/deployment-maps/DeploymentMapTargetController.js]: The best thing here, would be if we had drop downs for available roles... but that is totally out of scope and I appreciate that. 

Do you think it's worth changing this to `!!vm.target.ASG.LaunchConfig.PuppetRole.trim()`?

**Comment by trainb0t** [comment]
@jake-cross I have added some webdriver io tests for this feature. The slack channel is now also a link.
